### PR TITLE
ARROW-13944: [C++] Bump xsimd to latest version

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

**Comment by cyb70289** [comment]
Reported xsimd msvc build failure https://github.com/xtensor-stack/xsimd/issues/561

**Comment by cyb70289** [comment]
Hmm..., there are more issues need be fixed in xsimd.

File: cpp/cmake_modules/SetupCxxFlags.cmake [hunk 1]
```diff
@@ -618,6 +618,15 @@ endif()
 
 message(STATUS "Build Type: ${CMAKE_BUILD_TYPE}")
 
+# Workaround https://github.com/xtensor-stack/xsimd/issues/573
```
Comment (cyb70289) [inline cpp/cmake_modules/SetupCxxFlags.cmake:621]: To workaround xsimd build error if gcc -O0 on arm64. Disable xsimd if gcc & neon & debug build.
Details at https://github.com/xtensor-stack/xsimd/issues/573. This looks to me a compiler defect.

**Comment by cyb70289** [comment]
@ursabot please benchmark

File: cpp/cmake_modules/SetupCxxFlags.cmake (hunk 1 above)
Comment (pitrou) [reply cpp/cmake_modules/SetupCxxFlags.cmake:621]: Cool, let's wait for it then?

Comment (cyb70289) [reply cpp/cmake_modules/SetupCxxFlags.cmake:621]: Yes.

**Review by pitrou — APPROVED** [review]
+1. Thank you @cyb70289.

**Pull request closed without merging**